x/globalfee/types/params.go

```diff
@@ -32,11 +35,65 @@ func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 	}
 }
 
+// this requires the fee non-negative
```

Comment:
presently, when setup globalfee, in genesis or gov propose to write to store, the global fee must be sorted by denom. otherwise, panic.
if we do not require user to pay attention to denom order when setup global fee, 
we can sort it when call 
`func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {}`
```
func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
	p.MinimumGasPrices = p.GetMinimumGasPrices().Sort()
	return paramtypes.ParamSetPairs{
		paramtypes.NewParamSetPair(
			ParamStoreKeyMinGasPrices, &p.MinimumGasPrices, validateMinimumGasPrices,
		),
	}
}
```
however, the  *Param 's   MinimumGasPrices  might pass or fail `func (p Params) ValidateBasic() error ` depends on the validation is before or after the ` ParamSetPairs() `.
we can either sort it in `ValidateBasic()` again or leave it to users to pay attention of denom order when setup global fee.

Reply:
let's sort it for them.
i'd like to consider making the validation accept 0denom as an acceptable scenario though

Reply:
if we sort them, we sort them in two places, one is in `ParamSetPairs()` one is in `ValidateBasic()`, so maybe we call `ValidateBasic()` -> sortAndvalidateBasics()`

Reply:
I just do not know if we automatically sort them, if directly change genesis with unsorted globalfee,
 our validation will sort them and say its valid genesis. write to the db is not a problem, because we will sort them and write to store.
just do not know if in genesis, globalfee not sorted, if there is potential issues?